src/Storages/StorageMergeTree.h

```diff
@@ -145,6 +145,14 @@ class StorageMergeTree final : public MergeTreeData
     /// This set have to be used with `currently_processing_in_background_mutex`.
     DataParts currently_merging_mutating_parts;
 
+    /// Accessed under `currently_processing_in_background_mutex` lock.
```

Comment:
Consider using [TSA annotations](https://github.com/ClickHouse/ClickHouse/blob/cedf75ed5ebf2912490200ee82b6a110afc7bb70/base/base/defines.h#L137-L147)

Reply:
Wow, cool!

Reply:
@tavplubix it does not quite fit in this case because there is (at least) one usage of `unique_lock` instead of `lock_guard` in `selectPartsToMerge()` and we would not be able to build complete tree of function calls properly annotated.

https://lists.llvm.org/pipermail/cfe-dev/2016-November/051468.html

Reply:
You can use `TSA_SUPPRESS_WARNING_FOR_READ` in `selectPartsToMerge`

Reply:
This is impossible because `std::unique_lock` has no thread annotations. There is nothing to suppress, however, but it is needed to mark mutex as locked for Clang instead. It would work if we only used `std::lock_guard`.

Reply:
It is possible and it works fine with `std::unique_lock` and `TSA_SUPPRESS_WARNING_FOR_READ`